addons/sale_project/models/project.py

```diff
@@ -27,13 +27,25 @@ class Project(models.Model):
     has_any_so_with_nothing_to_invoice = fields.Boolean('Has a SO with an invoice status of No', compute='_compute_has_any_so_with_nothing_to_invoice')
     invoice_count = fields.Integer(related='analytic_account_id.invoice_count', groups='account.group_account_readonly')
     vendor_bill_count = fields.Integer(related='analytic_account_id.vendor_bill_count', groups='account.group_account_readonly')
+    partner_id = fields.Many2one(compute="_compute_partner_id", store=True, readonly=False)
+    should_display_partner = fields.Boolean("Should Display Customer", compute="_compute_should_display_partner")
 
     @api.model
     def _map_tasks_default_valeus(self, task, project):
         defaults = super()._map_tasks_default_valeus(task, project)
         defaults['sale_line_id'] = False
         return defaults
 
+    @api.depends('allow_billable')
+    def _compute_partner_id(self):
+        self.filtered(lambda project: not project.allow_billable).partner_id = False
+
+    @api.depends('allow_billable')
+    def _compute_should_display_partner(self):
+        billable_projects = self.filtered('allow_billable')
+        billable_projects.should_display_partner = True
+        (self - billable_projects).should_display_partner = False
```

Comment:
I think we could use `allow_billable` field instead of creating a field to know if we can display or not another field.